API Google Analytics/HelloAnalytics.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+
+from apiclient.discovery import build
+from oauth2client.service_account import ServiceAccountCredentials
+
+import httplib2
+from oauth2client import client
+from oauth2client import file
+from oauth2client import tools
+
+SCOPES = ['https://www.googleapis.com/auth/analytics.readonly']
+DISCOVERY_URI = ('https://analyticsreporting.googleapis.com/$discovery/rest')
+KEY_FILE_LOCATION = 'client_secrets.p12'
+SERVICE_ACCOUNT_EMAIL = ''
+VIEW_ID = ''
+
+def initialize_analyticsreporting():
+  """Initializes an analyticsreporting service object.
+
+  Returns:
+    analytics an authorized analyticsreporting service object.
+  """
+
+  credentials = ServiceAccountCredentials.from_p12_keyfile(
+    SERVICE_ACCOUNT_EMAIL, KEY_FILE_LOCATION, scopes=SCOPES)
+
+  http = credentials.authorize(httplib2.Http())
+
+  # Build the service object.
+  analytics = build('analytics', 'v4', http=http, discoveryServiceUrl=DISCOVERY_URI)
+
+  return analytics
+
+def get_report(analytics):
+    """ Use the Analytics Service Object to query the Analytics Reporting API V4.
+
+    Inputs:
+    	The analytics object contructed.
+    Returns:
+    	The dictionary with all the responses to the request
+    """
+    return analytics.reports().batchGet(
+    body={
+    "reportRequests": [
+    {
+    "viewId": "XXXX",
+    "dateRanges": [
+    {
+    "startDate": "30daysAgo",
+    "endDate": "yesterday"
+    }
+    ],
+    "metrics": [
+    {
+    "expression": "ga:newUsers"
+    },
+    {
+    "expression": "ga:sessions"
+    },
+    {
+    "expression": "ga:adClicks"
+    },
+    {
+    "expression": "ga:adCost"
+    },
+    {
+    "expression": "ga:CPM"
+    },
+    {
+    "expression": "ga:adClicks"
+    },
+    {
+    "expression": "ga:ROAS"
+    },
+    {
+    "expression": "ga:pageValue"
+    },
+    {
+    "expression": "ga:pageviews"
+    }
+    ],
+    "dimensions": [
+    {
+    "name": "ga:campaign"
+    }
+    ]
+    }
+    ]}).execute()
+
```

Comment:
This part is confusing me. Can you write this part a little more standardized? Besides, I think it's more straight forward defining the dict and then write the return statement. Besides, as I said earlier, this function can generate some exception too, we'd better raise them.

Reply:
If I write the dict outise the return statement I got an error:
return analytics.reports().batchGet(body).execute()
TypeError: method() takes 1 positional argument but 2 were given

The only way it works is if I write it inside the return statement, or I'm missing something.